app/src/main/java/com/example/paritosh/stormy/WeatherDataProvider.java

```diff
@@ -78,7 +105,8 @@ private CurrentWeather getCurrentDetails(String jsonData) throws JSONException {
     }
 
     public interface OnWeatherApiResponse {
-        public void onSuccess(CurrentWeather weather);
+        public void onSuccess(CurrentWeather weather, ArrayList<Hourly> hourly);
```

Comment:
hourly forecasts are a part of CurrentWeather and should be included within CurrentWeather. So, change the method signature to onSuccess(CurrentWeather weather). Also, you don't have to write public for a method in an interface.